src/server/game/AI/SmartScripts/SmartScript.cpp

```diff
@@ -1519,7 +1519,10 @@ void SmartScript::ProcessAction(SmartScriptHolder& e, Unit* unit, uint32 var0, u
                     if (TransportBase* trans = me->GetDirectTransport())
                         trans->CalculatePassengerPosition(dest.x, dest.y, dest.z);
 
-                me->GetMotionMaster()->MovePoint(e.action.MoveToPos.pointId, dest.x, dest.y, dest.z, e.action.MoveToPos.disablePathfinding == 0);
+                if (e.GetTargetType() == SMART_TARGET_HOME_POSITION)
+                    me->GetMotionMaster()->MoveTargetedHome();
```

Comment:
now, you calculate transport passenger position and dont need it, maybe the best would be a big switch case